From 83d2a4a9c5e353b44f7d918a15b9a1f4dc16b3f6 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 10 Jul 2024 09:16:37 +0300
Subject: [PATCH] [FSW-7134] fs1: boot_desc: parse boot desc struct

Take the necessary fields from the boot descriptor structure and store
for a local use

Change-Id: I3c327b04270faa651a83859b79a6d321bd1ff828
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index b1b7807..4aaf56a 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -21,6 +21,7 @@
 #include <efuse.h>
 #include <hl_fw_upd_if.h>
 #include <boot_desc.h>
+#include <string.h>
 #include "dev_uart.h"
 #include "progress_bits.h"
 
@@ -335,11 +336,17 @@
 	return true;
 }
 
-uint32_t boot_desc_copy_desc(__unused struct boot_desc *desc,
-			     __unused struct hl_fw_img_desc *local_img_desc,
+uint32_t boot_desc_copy_desc(struct boot_desc *desc,
+			     struct hl_fw_img_desc *local_img_desc,
 			     __unused enum fw_post_reset_cmd upd_status)
 {
-	/* TODO: implement for FS1 */
+	desc->spi_offset = local_img_desc->offset;
+	desc->img_size = local_img_desc->size;
+	desc->loadaddr = ((uint64_t)local_img_desc->load_high << 32) | local_img_desc->load_low;
+	desc->desc_ver = local_img_desc->desc_ver;
+	desc->start_addr = ((uint64_t)local_img_desc->start_high << 32) | local_img_desc->start_low;
+	desc->crc32 = local_img_desc->crc32;
+	memcpy(desc->chksum, local_img_desc->chksum, ARRAY_SIZE(desc->chksum));
 
 	return 0;
 }
diff --git a/common/boot_desc.h b/common/boot_desc.h
index 87fa831..116b691 100644
--- a/common/boot_desc.h
+++ b/common/boot_desc.h
@@ -23,6 +23,7 @@
 	uint32_t pub_key_size;
 	uint32_t sig_size;
 	uint64_t start_addr;
+	uint32_t crc32;
 };
 
 int boot_desc_init(enum hl_fw_boot_ptr_imgs img);
